code/codegen/src/main/kotlin/com/android/gradle/replicator/resgen/util/PregenFileSelector.kt

```diff
@@ -51,6 +65,35 @@ fun getRandomResource(random: Random, type: FileTypes, resourceQualifiers: List<
     return if (filteredFiles.isEmpty()) allFiles.sorted().random(random) else filteredFiles.sorted().random(random)
 }
 
+fun getResourceClosestToSize(type: FileTypes, size: Long): String? {
+    if (fileSizeCache[type] == null) {
+        // Calculate cache
+        val allFiles = listResourceFiles(getFolderFromType(type))
+
+        fileSizeCache[type] = mutableListOf()
+
+        val loader = Thread.currentThread().contextClassLoader
+        allFiles.forEach { fileName ->
+            // getContentLengthLong gets the uncompressed file size
+            fileSizeCache[type]!!.add(
```

Comment:
if you used a sorted data structure sorted on the file size, it would make the look up later a lot easier and faster, rather than going through the entire list.

Reply:
I think this will add some unnecessary complexity to the code. This is a list of the pre-generated resources, so it has somewhere around 2-20 elements, so it's not a very big deal to look at all the elements in the list.